Newsletter 96 (2020-05-06) (#395)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2020-05-06-newsletter.md

---
title: 'Bitcoin Optech Newsletter #96'
permalink: /en/newsletters/2020/05/06/
name: 2020-05-06-newsletter
slug: 2020-05-06-newsletter
type: newsletter
layout: newsletter
lang: en
---
This week's newsletter links to a discussion about using enhanced QR
codes for communicating large transactions, includes a field report from
Suredbits about building a high-availability LN node, and briefly
summarizes several recently transcribed talks and conversations.  Also
included are our regular sections with releases, release candidates, and
notable code changes from popular Bitcoin infrastructure software.

## Action items

{:.center style="font-size: 1.5em"}
<!-- $$\frac{50}{2^{\lfloor height/210000 \rfloor}}$$ -->
![Happy halving!](/img/posts/2020-05-halving.png)

## News

- **QR codes for large transactions:** QR codes can practically contain
  up to about 3 kilobytes of data, which is enough to accommodate
  typical user transactions but far too small to contain the largest

>   transactions users can send normally (about 400 kilobytes).  Riccardo

Reviewer: ```suggestion
  transactions users can normally send (about 400 kilobytes).  Riccardo
```

  Casatta and Christopher Allen each posted to the Bitcoin-Dev mailing
  list ([1][casatta qr], [2][allen qr]) a request for discussion
  hopefully leading to standardization of a method for visual
  communication of [Partially Signed Bitcoin Transactions][topic psbt]
  (PSBTs) and other potentially large data blobs related to Bitcoin
  wallet interaction.  See the [previous discussion][qr old] in the
  Specter DIY repository and [continue the discussion][qr new] in the
  Airgapped Signing repository.

## Field report: running a Lightning node in an enterprise environment

{% include articles/suredbits-enterprise-ln.md %}

## Recently transcribed talks and conversations

*[Bitcoin Transcripts][] is the home for transcripts of technical

> Bitcoin presentations and discussions. In this new <!-- TODO: remove "new" next month --> monthly feature we

Reviewer: ```suggestion
Bitcoin presentations and discussions. In this new <!-- TODO: remove "new" next month --> monthly feature, we
```

  be assessed according to five goals: reliability, timeliness,
  accessibility, privacy, and upgradability. She discussed the dangers of
  network partitions and [eclipse attacks][topic eclipse attacks]---and then explained why
  block-relay-only connections and anchor nodes are effective
  mitigations.  ([transcript][attacking xs], [video][attacking vid])

- **LND v0.10** Laolu Osuntokun, Joost Jager, and Oliver Gugger
  presented in virtual reality at Reckless VR. Osuntokun presented Tor
  and RPC enhancements in the latest release of LND plus a new channel
  feature called [anchor outputs][topic anchor outputs] that addresses
  the challenge of estimating onchain fees months in advance.
  Jager discussed the challenges of [multipart payments][topic multipath
  payments] including the splitting algorithm, what happens when the
  shards of the payment arrive at different times, and strategies for
  when a multipart payment fails. Gugger ended by discussing Partially
  Signed Bitcoin Transaction ([PSBT][topic psbt]) channel funding and
  the channel abstraction work that made this possible.
  ([transcript][lnd10 xs], [video][lnd10 vid])

> - **Grokking Bitcoin** Kalle Rosenbaum participated in a [Socratic
>   Seminar][] at London Bitcoin Devs one week and gave a presentation the
>   next week.  The seminar focused on the role of Bitcoin technical

Reviewer: This lead is a little awkward. He is a slight improvement though not my favorite:
```suggestion
- **Grokking Bitcoin** Kalle Rosenbaum led a [Socratic
  Seminar][] at London Bitcoin Devs and gave a presentation the
  following week.  The seminar focused on the role of Bitcoin technical
```

Author: I'm not sure Kalle led the seminar (the transcript is unattributed for privacy, but it seemed to me that someone else was leading).

  [partially-signed bitcoin transactions][topic psbt]).

    By default, descriptor wallets use the popular [BIP32][] HD wallet
    paths specified by BIPs [44][BIP44], [49][BIP49], and [84][BIP84]
    rather than the non-standardized fully-hardened path used in legacy
    Bitcoin Core HD wallets.  A number of wallet RPCs are unavailable
    with descriptor wallets, either because they don't make sense with
    descriptors or because developers are still adapting them to new
    edge cases.  The merge of this PR early in the 0.21 development
    cycle and the decision to make descriptor wallets a non-default
    option will give the new features six months to mature before
    their expected release.

- [Bitcoin Core #18038][] P2P: Mempool tracks locally submitted
  transactions to improve wallet privacy FIXME:bitschmidty

- [BIPs #893][] makes several changes to the [BIP340][] specification of
  [schnorr][topic schnorr signatures] pubkeys and signatures, with
  related changes also being made to the [BIP341][] specification of
  [taproot][topic taproot].  The major changes include:

    - *Alternative x-only pubkey tiebreaker:* this changes which variant

>       of a public key to use when only the key’s x-coordinate is known,

Reviewer 2: suggest: "when only the x-coordinate of the key is known,"

      as previously discussed (see [Newsletter #83][news83 alt
      tiebreaker]).

>     - *Updated nonce generation recommendadions:* the suggested method

Reviewer: ```suggestion
    - *Updated nonce generation recommendations:* the suggested method
```

>     - *Updated nonce generation recommendadions:* the suggested method

Reviewer 2: s/recommendadions/recommendations

>     - *Updated nonce generation recommendadions:* the suggested method

Reviewer 3: s/srecommendadions/recommendations

      to use for generating a signature nonce has been updated to
      prevent implementation-specific vulnerabilities (see previous
      discussion in Newsletters [#83][news83 precomputed] and
      [#91][news91 power analysis]).

    - *Updated tagged hashes:* the tag prefixed to hash function input
      for schnorr signatures has been updated to deliberately break

>       compatibility with the old draft specification.  Otherwise

Reviewer: ```suggestion
      compatibility with the old draft specification.  Otherwise,
```

  displayed a [demo][simplicity demo] showing what you can do with
  Simplicity today.  ([transcript][simplicity xs], [video][simplicity
  vid], [slides][simplicity slides])

- **Attacking Bitcoin Core** Amiti Uttarwar presented at LA BitDevs.
  Uttarwar discussed how changes to Bitcoin’s peer-to-peer layer should
  be assessed according to five goals: reliability, timeliness,
  accessibility, privacy, and upgradability. She discussed the dangers of
  network partitions and [eclipse attacks][topic eclipse attacks]---and then explained why
  block-relay-only connections and anchor nodes are effective
  mitigations.  ([transcript][attacking xs], [video][attacking vid])

- **LND v0.10** Laolu Osuntokun, Joost Jager, and Oliver Gugger
  presented in virtual reality at Reckless VR. Osuntokun presented Tor
  and RPC enhancements in the latest release of LND plus a new channel
  feature called [anchor outputs][topic anchor outputs] that addresses
  the challenge of estimating onchain fees months in advance.
  Jager discussed the challenges of [multipart payments][topic multipath
  payments] including the splitting algorithm, what happens when the

>   shards of the payment arrive at different times, and strategies for

Reviewer 4: nit: "strategies for dealing with when a...."./ "strategies for handling ..."

  when a multipart payment fails. Gugger ended by discussing Partially
  Signed Bitcoin Transaction ([PSBT][topic psbt]) channel funding and
  the channel abstraction work that made this possible.
  ([transcript][lnd10 xs], [video][lnd10 vid])

> - **Grokking Bitcoin** Kalle Rosenbaum both participated in a [Socratic

Reviewer 4: nit: remove 'both'

>   Seminar][] and gave a presentation at London Bitcoin Devs.  The

Reviewer 4: I don't think this link would be very meaningful for readers. I suggest either removing the link entirely and rewording to "developer meetup" or linking to https://bitdevs.org/running-a-great-socratic-seminar/.

  for receiving [spontaneous payments][topic spontaneous payments], and
  contains several other new features and bug fixes. Additionally, both
  novice and experienced users are encouraged to read the project's new
  [FAQ][cl faq].

- [LND 0.10.0-beta][lnd 0.10.0-beta] is a major version release that
  adds support for sending [multipath payments][topic multipath
  payments], funding channels using an external wallet via [Partially
  Signed Bitcoin Transactions][topic psbt] (PSBTs), the ability to create
  invoices larger than 0.043 BTC, and several other new features and bug
  fixes.  Additionally, users may wish to read the new [Operational
  Safety documentation][lnd op safety].

- [Bitcoin Core 0.20.0rc1][bitcoin core 0.20.0] is a release candidate
  for the next major version of Bitcoin Core.

## Notable code and documentation changes

*Notable changes this week in [Bitcoin Core][bitcoin core repo],
[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
[libsecp256k1][libsecp256k1 repo], [Rust-Lightning][rust-lightning repo],
[Bitcoin Improvement Proposals (BIPs)][bips repo], and [Lightning
BOLTs][bolts repo].*

> *Note: the commits to Bitcoin Core mentioned below apply to its master

Reviewer 4: This boilerplate warning was added last week as well. What's the justification for doing this? Doesn't the same warning apply to commits for other repos?

Author: > This boilerplate warning was added last week as well.

Variations on the notice also appear in newsletters 9, 10, 11, 12, 13, 14, 37, 40, 41, 42, 43, 67, and 68.  (Search for "master" in any of those newsletters.)

> What's the justification for doing this?

Without the warning, I think people will expect the changes we describe to appear in the next Bitcoin Core release (e.g. 0.20 in a few weeks).

> Doesn't the same warning apply to commits for other repos?

No.  Both LND and C-Lightning RCs releases are just tags on their master branch.  I think that's also the case for Eclair, but I'm not sure.  Rust-Lightning has tags that GitHub shows as releases but there's no release notes and they're all version 0.0.x, so I've been assuming they're not real releases.  The other repos don't do releases.

In short, AFAICT, Bitcoin Core is the only project we monitor that branches from master prior to a major release.  (Indeed, I'm unable to think of any other free software project I've followed that does that.)

As an alternative to the warning, we could simply not report on Bitcoin Core commits to master during RC cycles, though that would leave us with a huge backlog when at the end of each cycle.

Reviewer 4: ah ok. Makes sense. Thanks!

    rather than the non-standardized fully-hardened path used in legacy
    Bitcoin Core HD wallets.  A number of wallet RPCs are unavailable
    with descriptor wallets, either because they don't make sense with
    descriptors or because developers are still adapting them to new
    edge cases.  The merge of this PR early in the 0.21 development
    cycle and the decision to make descriptor wallets a non-default
    option will give the new features six months to mature before
    their expected release.

- [Bitcoin Core #18038][] improves privacy when initially broadcasting a
  transaction by reducing the frequency that a wallet will attempt to resend
  from about 30 minutes to about once a day. Previously, an entity surveilling
  the network could see multiple broadcasts of the same transaction during these
  resend periods from the same node and conclude that wallet was the originator.
  By reducing the frequency of resending attempts from the wallet, the
  originator of the transaction has less chance of being identified.

  To ensure new transactions reach the network even without frequent
  indiscriminate rebroadcast by the wallet, this PR also adds a type of

>   "unbroadcast" transaction within the mempool. An unbroadcasted transaction is

Reviewer 4: nit: we usually introduce new terms with _emphasis_ rather than "quotes"

Reviewer 4: nit: We should use unbroadcast, rather than unbroadcasted, as the past participle/adjective here (which is, after all, what the new feature is named in Bitcoin Core). That would make this sentence "An unbroadcast transcation is ..."

Reviewer 3: Your changes LGTM @jnewbery , thanks!

    option will give the new features six months to mature before
    their expected release.

- [Bitcoin Core #18038][] improves privacy when initially broadcasting a
  transaction by reducing the frequency that a wallet will attempt to resend

>   from about 30 minutes to about once a day. Previously, an entity surveilling

Reviewer 2: suggest s/about once a day/once every 12-36 hours/

Reviewer 4: I prefer 'about once a day'. The exact distribution seems too low-level for a summary write-up, and people who are interested can click through to the PR for full details.
